packages/sign-client/src/controllers/engine.ts

```diff
@@ -90,6 +99,11 @@ export class Engine extends IEngine {
       this.registerExpirerEvents();
       this.client.core.pairing.register({ methods: Object.keys(ENGINE_RPC_OPTS) });
       this.initialized = true;
+
+      setTimeout(() => {
```

Comment:
Context on why we need a timeout here?

Reply:
Timeout is needed to allow wallets to subscribe to `session_event` after `SignClient.init` resolves. The 1s delay is an extra precaution so the first emitted event is not missed due to app not being in a fully ready state to handle events. We can adjust though